packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,400 @@
+ * explicitly stated, will be preprocessed with whatever value they return (assuming that value is serializable). If a value cannot be preprocessed for whatever reason,
+ * null is returned.
+ *
+ *
+ * NOTE: this function recursively calls itself to preprocess a log
+ *
+ * @param {any} props a generic variable that represents a value that needs to be preprocessed before being sent through postMessage().
+ * @param {boolean} [attemptToSerializeFunctions=false] - Whether or not the function should attempt to preprocess a function by invoking it. USE WITH CAUTION!
+ * @returns {any} the serializable version of the generic.
+ */
+export const preprocessLogLikeForSerialization = (props, attemptToSerializeFunctions = false) => {
+  try {
+    if (_.isArray(props)) {
+      return props.map((prop) => {
+        try {
+          return preprocessLogLikeForSerialization(prop, attemptToSerializeFunctions)
+        } catch (e) {
+          return null
+        }
+      })
+    }
+
+    if (_.isPlainObject(props)) {
+      // only attempt to try and serialize dom elements and function (if attemptToSerializeFunctions is set to true)
+      let objWithPossiblySerializableProps = _.pickBy(props, (value) => {
+        const isSerializable = isSerializableInCurrentBrowser(value)
+
+        if (!isSerializable && $dom.isDom(value) || _.isFunction(value) || _.isObject(value)) {
+          return true
+        }
+
+        return false
+      })
+
+      let objWithOnlySerializableProps = _.pickBy(props, (value) => isSerializableInCurrentBrowser(value))
+
+      // assign the properties we know we can serialize here
+      let preprocessed: any = preprocessForSerialization(objWithOnlySerializableProps)
+
+      // and attempt to serialize possibly unserializable props here and fail gracefully if unsuccessful
+      _.forIn(objWithPossiblySerializableProps, (value, key) => {
+        try {
+          preprocessed[key] = preprocessLogLikeForSerialization(value, attemptToSerializeFunctions)
+        } catch (e) {
+          preprocessed[key] = null
+        }
+      })
+
+      return preprocessed
+    }
+
+    if ($dom.isDom(props)) {
+      if (props.length !== undefined && $dom.isJquery(props)) {
+        const serializableArray: any[] = []
+
+        // in the case we are dealing with a jQuery array, preprocess to a native array to nuke any prevObject(s) or unserializable values
+        props.each((key) => serializableArray.push(preprocessLogLikeForSerialization(props[key], attemptToSerializeFunctions)))
+
+        return serializableArray
+      }
+
+      // otherwise, preprocess the element to an object with pertinent DOM properties
+      const serializableDOM = preprocessDomElement(props)
+
+      return serializableDOM
+    }
+
+    /**
+     * When preprocessing a log, there might be certain functions we want to attempt to serialize.
+     * One of these instances is the 'table' key in consoleProps, which has contents that CAN be serialized.
```

Comment:
renderProps, consoleProps and consoleProps.table correct?

Reply:
In this case, just `consoleProps.table`. `renderProps` and `consoleProps` are typically functions, but when emitted to the `log:added` or `log:changed` events their functions are [invoked](https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/cypress/log.ts#L274). Since the same behavior happens on the front end, we can keep `consoleProps` and `renderProps` as objects due to the details of the [invoke](https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/cypress/log.ts#L258) function. Ultimately the same thing shows up in the reporter.

Reply:
I also forgot to mention that I added renderProps assertions in [3a1bc84](https://github.com/cypress-io/cypress/pull/20949/commits/3a1bc844617f7a436e362ea1d3ab99b9ac43e916) for `cy.request` to be sure. We can't use `cy.session` `cy.route` or `cy.server` within `cy.origin` , so I _think_ this is the only command with actual renderProps coming from the spec bridge